src/index.js

```diff
@@ -289,7 +289,11 @@ class Libp2p extends EventEmitter {
     const dialable = Dialer.getDialable(peer)
     let connection
     if (PeerInfo.isPeerInfo(dialable)) {
```

Comment:
This technically works because PeerInfo is duck typing it's check, but that's not the case everywhere which had me second guessing this.

The whole `getDialable` logic is giving me pause, especially with the PeerStore changes. I feel like we should just be giving an array of multiaddrs to the dialer to dial. This would tie in nicely with adding multiaddr confidence in the future, as the dialer could just be given a prioritized array.

I feel like the logic here should change a bit:
1. Ensure the peer/multiaddr `dialProtocol` was given is in the PeerStore.
2. Get the addresses from the AddressBook for this peer.
3. If the provided `peer` was an address, make it first in the multiaddr array that will be dialed.
4. Call dial with the multiaddr array

I think we should add 2 utilities to PeerStore until we fully deprecate PeerInfo.
1. A method to take a PeerInfo instance and store that data in the PeerStore
2. A method to aggregate from all the PeerStore books for a PeerId into a PeerInfo and return that
This could simplify a lot of the logic scattered around the repo that has to account for PeerInfo eventually going away.

Reply:
> The whole getDialable logic is giving me pause, especially with the PeerStore changes. I feel like we should just be giving an array of multiaddrs to the dialer to dial. This would tie in nicely with adding multiaddr confidence in the future, as the dialer could just be given a prioritized array.

I also want to reformulate the dialer in the `getDialable` logic. I was postponing this change to go together with `peer-info`, as deprecating `peer-info` will create the need for that change.

> I feel like the logic here should c hange a bit:
> 
> Ensure the peer/multiaddr dialProtocol was given is in the PeerStore.
> Get the addresses from the AddressBook for this peer.
> If the provided peer was an address, make it first in the multiaddr array that will be dialed.
> Call dial with the multiaddr array

Yes, I was thinking on something like this! 🚀 

> I think we should add 2 utilities to PeerStore until we fully deprecate PeerInfo.
>
> A method to take a PeerInfo instance and store that data in the PeerStore

I currently have `peerStore.put()` which does that. It is an adapter for the breaking change.

> A method to aggregate from all the PeerStore books for a PeerId into a PeerInfo and return that
> This could simplify a lot of the logic scattered around the repo that has to account for PeerInfo eventually going away.

`peerStore.peers` getter currently does that, until we do the breaking change.

Reply:
So, should we do this together with `peer-info` deprecation?

The idea is to get it all together in the same final release, so I think it would make the development easier

Reply:
Yeah, if the plan is to release everything in 0.28, we can add this change to that PR.